examples/aegp/Justfile

```diff
@@ -0,0 +1,81 @@
+PluginName       := "AegpDemo"
+BundleIdentifier := "com.adobe.AfterEffects.AegpDemo"
+BinaryName       := lowercase(PluginName)
+
+set windows-shell := ["powershell.exe", "-NoLogo", "-Command"]
+
+TargetDir := env_var_or_default("CARGO_TARGET_DIR", "target")
+export AESDK_ROOT := if env("AESDK_ROOT", "") == "" { justfile_directory() / "../../../../sdk/AfterEffectsSDK" } else { env_var("AESDK_ROOT") }
+export PRSDK_ROOT := if env("PRSDK_ROOT", "") == "" { justfile_directory() / "../../../../sdk/Premiere Pro 22.0 C++ SDK" } else { env_var("PRSDK_ROOT") }
+
+[windows]
+build:
+    cargo build
+    if (-not $env:NO_INSTALL) { \
+        Start-Process PowerShell -Verb runAs -ArgumentList "-command Set-Location '{{source_directory()}}'; Copy-Item -Force '{{TargetDir}}\debug\{{BinaryName}}.dll' 'C:\Program Files\Adobe\Common\Plug-ins\7.0\MediaCore\{{PluginName}}.aex'" \
+    }
+
+[windows]
+release:
+    cargo build --release
+    Copy-Item -Force '{{TargetDir}}\release\{{BinaryName}}.dll' '{{TargetDir}}\release\{{PluginName}}.aex'
+    if (-not $env:NO_INSTALL) { \
+        Start-Process PowerShell -Verb runAs -ArgumentList "-command Set-Location '{{source_directory()}}'; Copy-Item -Force '{{TargetDir}}\release\{{BinaryName}}.dll' 'C:\Program Files\Adobe\Common\Plug-ins\7.0\MediaCore\{{PluginName}}.aex'" \
+    }
+
+[macos]
+build:
+    cargo build
+    just -f {{justfile()}} create_bundle debug {{TargetDir}}
+
+[macos]
+release:
+    cargo build --release
+    just -f {{justfile()}} create_bundle release {{TargetDir}}
+
+[macos]
+create_bundle profile TargetDir:
+    #!/bin/bash
+    set -e
+    echo "Creating plugin bundle"
+    rm -Rf {{TargetDir}}/{{profile}}/{{PluginName}}.plugin
+    mkdir -p {{TargetDir}}/{{profile}}/{{PluginName}}.plugin/Contents/Resources
+    mkdir -p {{TargetDir}}/{{profile}}/{{PluginName}}.plugin/Contents/MacOS
+
+    echo "AEgxFXTC" >> {{TargetDir}}/{{profile}}/{{PluginName}}.plugin/Contents/PkgInfo
+    /usr/libexec/PlistBuddy -c 'add CFBundlePackageType string AEgx' {{TargetDir}}/{{profile}}/{{PluginName}}.plugin/Contents/Info.plist
+    /usr/libexec/PlistBuddy -c 'add CFBundleSignature string FXTC' {{TargetDir}}/{{profile}}/{{PluginName}}.plugin/Contents/Info.plist
+    /usr/libexec/PlistBuddy -c 'add CFBundleIdentifier string {{BundleIdentifier}}' {{TargetDir}}/{{profile}}/{{PluginName}}.plugin/Contents/Info.plist

```

Comment:
Note that I have duplicated the justfile almost exactly, AEGPs require slight differences on macos in their plist files, this is ugly and I would prefer not to repeat this script.

Reply:
We could create the `PkgInfo` file in `build.rs` pipl code since it knows that the effect is AEGP or regular. Then in Justfile we would just copy that file. This way everything will be automatic